bin/proving-ground.js

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env node
+'use strict';
+
+const path = require('path');
+const provingGround = require('../');
+
+const argv = require('yargs')
+  .usage('Usage: $0 [options] [files or directories]')
+  .option('e', {
+    alias: 'exec',
+    default: 'node',
+    describe: 'the command to pass to prove as its `exec` parameter',
+    type: 'string'
+  })
+  .option('n', {
+    alias: 'num-processes',
+    default: 1,
+    describe: 'the level of parallelism `prove` should employ',
+    type: 'string'
+  })
+  .option('before', {
+    default: null,
+    describe: 'a module exporting a function to run before `prove`',
+    type: 'string'
+  })
+  .option('after', {
+    default: null,
+    describe: 'a module exporting a function to run after `prove`',
+    type: 'string'
+  })
+  .help('h')
+  .alias('h', 'help')
+  .version()
+  .argv;
+
+const config = {
+  exec: argv.exec,
+  numProcesses: argv.n,
+  files: argv._
+};
+
+if (argv.before !== null) {
+  config.before = require(path.resolve(argv.before));
```

Comment:
the idea being that the CLI will be invoked in another node project with the before / after modules as local modules?

I think that's reasonable but the semantics could be confusing, eg: the following will act the same even though node devs familiar with the module loader would probably expect them to work differently:
```
proving-ground --before some-module tests/

proving-ground --before ./some-module tests/
```

I think that's reasonable, just talking out loud. The dev could easily use the relative path into `node_modules` if need be.